src/components/QrView.tsx

```diff
@@ -19,14 +19,12 @@ interface Props {
 
 const QrComponent = ({ navigateForward }: Props) => {
 	const [hasCameraPermission, setCameraPermission] = useState('null');
-	const [scanned, setScanned] = useState(false);
 
 	const { state, dispatch } = useContext(Ctx);
```

Comment:
Why do we use application-level state for this?

Reply:
Because many screens will need the ability to go back to "scan QR code". I tested myself that if we go (navigate) back to the QR screen from another screen, the QR screen isn't reloaded, so the local component level "scanned" state is still true, and the scanner will not scan if "scanned" is true. Putting setScannned(false) in useEffect won't work since the QR component is not reloaded, so it is the job of the other screen to update scanned.

Reply:
Wait ignore my approval idk how to remove it. Actually, I don't think it would be good to get other screens to set scanned back to false. I'm sure there's probably a way to force run the react hook constructor again. Or if you really want to do it this way, would you not need to put it in context instead?